ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDTFGetSplits.java

```diff
@@ -307,6 +309,17 @@ public PlanFragment createPlanFragment(String query, int num, ApplicationId spli
         needsCleanup = false;
       }
 
+      // Pass the ValidTxnList and ValidTxnWriteIdList snapshot configurations corresponding to the input query
+      HiveConf driverConf = driver.getConf();
+      String validTxnString = driverConf.get(ValidTxnList.VALID_TXNS_KEY);
+      if (validTxnString != null) {
+        jc.set(ValidTxnList.VALID_TXNS_KEY, validTxnString);
+      }
+      String validWriteIdString = driverConf.get(ValidTxnWriteIdList.VALID_TABLES_WRITEIDS_KEY);
+      if (validWriteIdString != null) {
+        jc.set(ValidTxnWriteIdList.VALID_TABLES_WRITEIDS_KEY, validWriteIdString);
+      }
+
```

Comment:
can we get a unit test for this? which can be as simple as a check for the presence of the configurations on jobconf for now.

Reply:
We already have TestAcidOnTez.testGetSplitsLocks which covers this.